private/serialization.nim

```diff
@@ -62,12 +62,40 @@ proc representObject*(value: string, ts: TagStyle,
   ## represents a string as YAML scalar
   c.put(scalarEvent(value, tag, yAnchorNone))
 
+proc parseHex[T: int8|int16|int32|int64|uint8|uint16|uint32|uint64](s: string): T =
+  result = 0
+  var i = 2
+  while true:
+    case s[i]
+    of '_': discard
+    of '0'..'9': result = result shl 4 or (T(ord(s[i]) - ord('0')) and 0xFF)
+    of 'a'..'f': result = result shl 4 or (T(ord(s[i]) - ord('a') + 10) and 0xFF)
+    of 'A'..'F': result = result shl 4 or (T(ord(s[i]) - ord('A') + 10) and 0xFF)
+    of '\0': break
+    else: raise newException(ValueError, "Invalid character in hex: " & s[i])
```

Comment:
Please use `escape("" & s[i])` here so that non-printable characters are displayed properly.